### PR TITLE
[xray] separate limits for max cards in section and max total cards

### PR DESCRIPTION
**Pull request by sansolovyov**
Improves situation in #46346 after what #45909 had done.

### CODE REVIEW

File: src/metabase/xrays/automagic_dashboards/core.clj [hunk 1]
```diff
@@ -560,7 +560,8 @@
            :cards dashcards)))
 
 (def ^:private ^:const ^Long max-related 8)
-(def ^:private ^:const ^Long max-cards 30)
+(def ^:private ^:const ^Long max-cards 15)
```
Comment (crisptrutski) [inline src/metabase/xrays/automagic_dashboards/core.clj:563]: I think you should rename this one too in order to clarify the context, e.g. `max-cards-section` perhaps.

Comment (piranha) [reply src/metabase/xrays/automagic_dashboards/core.clj:563]: I'm not too sure about that unfortunately. This is like a default limit for everything and I'm not understanding the code well enough.

File: src/metabase/xrays/automagic_dashboards/core.clj [hunk 2]
```diff
@@ -876,7 +877,8 @@
                                                :dashboard-templates-prefix ["table"]})
                                             opts)]
                            (automagic-dashboard root'))
-                         (let [root'     (merge root
+                         (let [opts      (assoc opts :show max-cards-total)
```
Comment (crisptrutski) [inline src/metabase/xrays/automagic_dashboards/core.clj:880]: I'm not very familiar with the x-ray code, but I find it pretty hard to follow that we have these two levels where `show` can be set to a constant. Do you mind giving a brief outline of how the the flow works for "section" versus "total"?

Also, should we not code  bit more defensively here, and do something like `(merge {:show max-cards-total} opts)` or `(update opts :show (fnil min max-cards-total) max-cards-total)`?

File: src/metabase/xrays/automagic_dashboards/core.clj (hunk 1 above)
Comment (crisptrutski) [reply src/metabase/xrays/automagic_dashboards/core.clj:563]: Fair enough - that makes two of us.

File: src/metabase/xrays/automagic_dashboards/core.clj (hunk 2 above)
Comment (piranha) [reply src/metabase/xrays/automagic_dashboards/core.clj:880]: If you remember, that was the place where it said `(assoc opts :show :all)`, so I'm just bringing back some override :)

**Review by crisptrutski — APPROVED** [review]
I don't understand this well enough, but seeing that its basically a revert to something more conservative LFG
